### PR TITLE
Bombers2 OK

### PR DESCRIPTION
**Pull request by louist103**
**Before** opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.c [hunk 1]
```diff
@@ -32,39 +35,404 @@ const ActorInit En_Bombers2_InitVars = {
...
+        case 4:
+            this->unk_28E = -0xFA0;
+            break;
+        case 1:
+        case 2:
+            this->unk_28E = -0x1770;
+            break;
+        default:
+            break;
+    }
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.c:215]: Same here

File: src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.c [hunk 2]
```diff
@@ -32,39 +35,404 @@ const ActorInit En_Bombers2_InitVars = {
...
+                    func_80C04B40(this);
+                    break;
+                case 2:
+                    this->textIdIndex = 3;
+                    this->actor.textId = sTextIds[this->textIdIndex];
+                    func_80151938(globalCtx, this->actor.textId);
+                    this->unk_2CE = 0xF;
+                    break;
+                case 3:
+                    for (j = 0; j < 5; j++) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.c:265]:
```suggestion
                    for (j = 0; j < ARRAY_COUNT(this->correctDigitSlots); j++) {
```

File: include/z64.h [hunk 3]
```diff
@@ -861,10 +861,8 @@ typedef struct {
     /* 0x12046 */ UNK_TYPE1 pad12046[0x2];
     /* 0x12048 */ u8 unk12048; // EnKakasi
     /* 0x12049 */ UNK_TYPE1 pad12049[0xB];
-    /* 0x12054 */ s16 unk12054; // First digit in lottery code guess
-    /* 0x12056 */ s16 unk12056; // Second digit lottery code guess
-    /* 0x12058 */ s16 unk12058; // Third digit lottery code guess
-    /* 0x1205A */ UNK_TYPE1 pad1205A[0x10];
+    /* 0x12054 */ s16 unk12054[5]; // Number entry
```
Comment (AngheloAlf) [inline include/z64.h:864]: What do you mean by "Number entry"?

File: src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.h [hunk 4]
```diff
@@ -3,15 +3,42 @@
...
     /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x140];
+    /* 0x014C */ SkelAnime skelAnime;
+    /* 0x0188 */ Vec3s morphTable[OBJECT_CS_LIMB_MAX];
+    /* 0x0206 */ Vec3s jointTable[OBJECT_CS_LIMB_MAX];
     /* 0x0284 */ EnBombers2ActionFunc actionFunc;
-    /* 0x0288 */ char unk_288[0x94];
+    /* 0x0288 */ s16 unk_288;
+    /* 0x028A */ s16 unk_28A;
+    /* 0x028C */ char pad_28C[2];
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.h:20]:
```suggestion
    /* 0x028C */ UNK_TYPE1 pad_28C[2];
```

File: src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.h [hunk 5]
```diff
@@ -3,15 +3,42 @@
...
+    /* 0x028E */ s16 unk_28E;
+    /* 0x0290 */ s16 unk_290;
+    /* 0x0292 */ char pad_292[4];
+    /* 0x0296 */ s16 unk_296;
+    /* 0x0298 */ char pad_298[4];
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.h:25]:
```suggestion
    /* 0x0292 */ UNK_TYPE1 pad_292[4];
    /* 0x0296 */ s16 unk_296;
    /* 0x0298 */ UNK_TYPE1 pad_298[4];
```

File: src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.c [hunk 6]
```diff
@@ -32,39 +35,403 @@ const ActorInit En_Bombers2_InitVars = {
...
+    this->unk_2C0 = 1;
+    this->actionFunc = func_80C04D8C;
+}
+
+void func_80C04D8C(EnBombers2* this, GlobalContext* globalCtx) {
+    Player* player = GET_PLAYER(globalCtx);
+    f32 temp = this->skelAnime.curFrame;
+
+    if ((this->unk_2AC == 0) && (this->lastAnimFrame <= temp)) {
+        if ((this->unk_2A8 & 1) == 0) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.c:197]:
```suggestion
        if (!(this->unk_2A8 & 1)) {
```

File: src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.c [hunk 7]
```diff
@@ -32,39 +35,403 @@ const ActorInit En_Bombers2_InitVars = {
...
+        rot->x += this->unk_296;
+    }
+    if (limbIndex == OBJECT_CS_LIMB_11) {
+        rot->x += this->unk_28A;
+        rot->z += this->unk_288;
+    }
+    if ((limbIndex == OBJECT_CS_LIMB_0F) || (limbIndex == OBJECT_CS_LIMB_13) || (limbIndex == OBJECT_CS_LIMB_14)) {
+        *dList = NULL;
+    }
+    return 0;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bombers2/z_en_bombers2.c:420]:
```suggestion
    return false;
```

File: src/overlays/actors/ovl_En_Bomjima/z_en_bomjima.c [hunk 8]
```diff
@@ -114,7 +114,7 @@ void EnBomjima_Init(Actor* thisx, GlobalContext* globalCtx) {
     this->actor.colChkInfo.mass = MASS_IMMOVABLE;
     ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 19.0f);
     this->actor.gravity = -3.0f;
-    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_cs_Skel_00F82C, &object_cs_Anim_0064B8, this->jointTable,
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_cs_Skel_00F82C, &gBomberIdleAnim, this->jointTable,
                        this->morphTable, 21);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bomjima/z_en_bomjima.c:118]:
```suggestion
                       this->morphTable, OBJECT_CS_LIMB_MAX);
```

File: src/overlays/actors/ovl_En_Bomjimb/z_en_bomjimb.c [hunk 9]
```diff
@@ -78,7 +78,7 @@ void EnBomjimb_Init(Actor* thisx, GlobalContext* globalCtx) {
     this->actor.colChkInfo.mass = MASS_IMMOVABLE;
     ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 19.0f);
     this->actor.gravity = -2.0f;
-    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_cs_Skel_00F82C, &object_cs_Anim_0064B8, this->jointTable,
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &object_cs_Skel_00F82C, &gBomberIdleAnim, this->jointTable,
                        this->morphTable, 21);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Bomjimb/z_en_bomjimb.c:82]:
```suggestion
                       this->morphTable, OBJECT_CS_LIMB_MAX);
```

**Review by hensldm — COMMENTED** [review]
Some minor things

**Review by AngheloAlf — APPROVED** [review]

**Review by hensldm — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**